wgengine/magicsock/magicsock.go

```diff
@@ -3186,6 +3292,37 @@ func (de *discoEndpoint) noteActiveLocked() {
 	}
 }
 
+// cliPing starts a ping for the "tailscale ping" command. res is value to call cb with,
+// already partially filled.
+func (de *discoEndpoint) cliPing(res *ipnstate.PingResult, cb func(*ipnstate.PingResult)) {
+	now := time.Now()
+
+	de.mu.Lock()
+	udpAddr, derpAddr := de.addrForSendLocked(now)
+	// Session active? If so, avoid sending a ping. They're already going out every
+	// 2 seconds. Just return what we know.
+	if !udpAddr.IsZero() && now.Before(de.trustBestAddrUntil) {
```

Comment:
Little nervous about this, in that it makes a debug command trust what our code is doing. If we're using the debug command, aren't we suspecting that our code is misbehaving?

I'd rather just send pings always, to be sure.

Reply:
Done.